src/portalbackend/PortalBackend.DBAccess/Repositories/ApplicationRepository.cs

```diff
@@ -404,16 +405,19 @@ public IQueryable<CompanyApplication> GetAllCompanyApplicationsDetailsQuery(stri
             .Select(ca => new { ca.ApplicationStatusId, ca.Company, ca.Company!.Address, ca.Company.CompanyIdentifiers })
             .Select(ca => new ClearinghouseData(
                 ca.ApplicationStatusId,
-                new ParticipantDetails(
+                ca.Company!.BusinessPartnerNumber!,
+                new LegalEntity(
                     ca.Company!.Name,
-                    ca.Address!.City,
-                    ca.Address.Streetname,
-                    ca.Company.BusinessPartnerNumber,
-                    ca.Address.Region,
-                    ca.Address.Zipcode,
-                    ca.Address.Country!.CountryLongNames.Where(cln => cln.ShortName == "en").Select(cln => cln.LongName).SingleOrDefault(),
-                    ca.Address.CountryAlpha2Code),
-                ca.CompanyIdentifiers.Select(ci => new UniqueIdData(ci.UniqueIdentifier!.Label, ci.Value))))
+                    new LegalAddress(
+                        ca.Address!.CountryAlpha2Code,
+                        string.Format("{0}-{1}", ca.Address!.CountryAlpha2Code, ca.Address!.Region),
+                        ca.Address.City,
+                        ca.Address.Zipcode,
+                        !string.IsNullOrEmpty(ca.Address.Streetnumber) ? string.Format("{0} {1}", ca.Address.Streetname, ca.Address.Streetnumber) : ca.Address.Streetname
```

Comment:
could you please switch for easier readability

```suggestion
                        string.IsNullOrEmpty(ca.Address.Streetnumber) ? ca.Address.Streetname : string.Format("{0} {1}", ca.Address.Streetname, ca.Address.Streetnumber) 
```

Reply:
done